src/extension/app.tsx

```diff
@@ -33,62 +33,49 @@ ReactDOM.render(
 let activeTabId = parseInt(location.href.match(/\?tab=([0-9]*$)/)[1]);
 chrome.runtime.sendMessage({ fetch: true }, function (response) {
     if (response.payloads) {
-        let payloads = response.payloads;
-        let size = 0;
-        let count = 0;
-        let structuredEvents = {};
-        let structuredSchemas = {};
-        let session = [];
-        let activeId;
+        let extensionPayloads = response.payloads;
         let activeIndex = 0;
+        let tabPayloads = {};
 
-        // Reconstruct uncompressed clarity payload
-        for (let entry of payloads) {
-            size += entry.length;
-            let json = JSON.parse(uncompress(entry.payload));
-            let tabId = entry.tabId;
-            let id = json.envelope.impressionId;
-            if (!(id in structuredEvents)) {
-                structuredEvents[id] = { envelope: json.envelope, events: [] };
-                structuredEvents[id]["envelope"].dateTime = entry.dateTime;
-                structuredEvents[id]["envelope"].summary = [];
-                structuredSchemas[id] = new clarity.converter.SchemaManager();
-                if (tabId === activeTabId) {
-                    activeId = id;
-                }
-            }
+        extensionPayloads.sort(compareExtensionPayloadsByTime);
```

Comment:
>compareExtensionPayloadsByTime [](start = 31, length = 30)

Is by time sufficient? What about sequence or event id?

Reply:
These are extension payloads that go from browser tab to the extension context - they are strictly sorted by time

---
In reply to: [179110351](https://github.com/Microsoft/clarity-tools/pull/38#discussion_r179110351) [](ancestors = 179110351)